juju/juju
# [JUJU-1944] DB-backed lease store implementation

Implementing leases in Dqlite requires a DB-backed implementation of the `lease.Store` indirection. 

Here we add such an implementation.

This constitutes the core functionality, which will be subject to review and iteration upon threading it into the lease manager worker.

## QA steps

No functional changes yet, but changes can be verified by running tests:
- Run `make dqlite-deps-pull`.
- Create a file in the project root, _test.list_, with the following contents:
```
github.com/juju/juju/database
github.com/juju/juju/worker/lease
```
- Run `TEST_PACKAGE_LIST=test.list make test` and ensure tests pass.

## Documentation changes

None.

## Bug reference

N/A
